### DISCUSSION TITLE
using `useSprings` in more of a "react" way

### DISCUSSION THREAD

**Discussion by d-pollard** [pmndrs/react-spring#1808] [Support]
Hi all!

Currently I'm struggling a bit as I'm trying to wrap my head around how I can use `useSprings` to manage a dynamic list of elements (can be added, removed, and deleted).

Currently, I am using `RefObject` to keep track of all the incoming items and where their ID is located at in the list of indexes. 

This works well until I need to delete or remove items from the list. In that case, I have a dead reference to an item that no longer exists.

Is there a way I could use `useSprings` by passing in unique IDs possibly instead of `index`? or maybe what I'm trying to do is simpler and you might have a suggestion for that?

**Comment by joshuaellis** [comment]
I would look at useTransition in this case. If you can provide a code sandbox, I can probably assist a little better!

**Reply by d-pollard** [reply]
hmm, would you use `useTransition` in conjunction with `useSprings` ?
